src/lib/resolvers/message.ts

```diff
@@ -32,11 +35,22 @@ async function resolveByLink(parameter: string, message: Message): Promise<Messa
 	const guild = container.client.guilds.cache.get(guildId as Snowflake);
 	if (guild !== message.guild) return null;
 
-	const channel = guild.channels.cache.get(channelId as Snowflake) as GuildBasedChannelTypes;
+	return getMessageFromChannel(channelId, messageId, message.author);
+}
+
+async function resolveByChannelAndMessage(parameter: string, message: Message): Promise<Message | null> {
+	const result = ChannelMessageRegex.exec(parameter)?.groups;
+	if (!result) return null;
+
+	return getMessageFromChannel(result.channelId, result.messageId, message.author);
+}
+
+async function getMessageFromChannel(channelId: Snowflake, messageId: Snowflake, originalAuthor: User): Promise<Message | null> {
+	const channel = container.client.channels.cache.get(channelId as Snowflake) as GuildBasedChannelTypes;
```

Comment:
```suggestion
	const channel = container.client.channels.cache.get(channelId) as GuildBasedChannelTypes;
```
*too lazy to open vscode and push, so I'll do it here*